domain/modelagent/state/modelstate_test.go

```diff
@@ -65,6 +74,85 @@ func (s *modelStateSuite) createMachineWithName(c *gc.C, name machine.Name) stri
 	return uuid.String()
 }
 
+// registerAgentBinary is a testing utility function that registers the fact
+// that an agent binary exists in the models store for the provided version. The
+// metadata for the newly created binary is returned to the caller upon creation.
+func (s *modelStateSuite) registerAgentBinary(
+	c *gc.C,
+	version coreagentbinary.Version,
+) coreagentbinary.Metadata {
+	runner := s.TxnRunner()
+
+	type objectStoreMeta struct {
+		UUID   string `db:"uuid"`
+		SHA256 string `db:"sha_256"`
+		SHA384 string `db:"sha_384"`
+		Size   int    `db:"size"`
+	}
+
+	storeUUID := uuid.MustNewUUID().String()
+	stmt, err := sqlair.Prepare(`
+INSERT INTO object_store_metadata (uuid, sha_256, sha_384, size)
+VALUES ($objectStoreMeta.uuid, $objectStoreMeta.sha_256, $objectStoreMeta.sha_384, $objectStoreMeta.size)
+`, objectStoreMeta{})
```

Comment:
nit: for tests we typically just use straight sql, is easier. not going to block on this